Content for speculative loading (#29183)
mdn/content: files/en-us/web/api/performancenavigationtiming/activationstart/index.md

---

> title: "PerformanceNavigationTiming: activationStart property"

Reviewer: Also needs to be added to https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming

Author: Ooops, missed that! Added now.

short-title: activationStart
slug: Web/API/PerformanceNavigationTiming/activationStart
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.PerformanceNavigationTiming.activationStart
---

{{APIRef("Performance API")}}{{SeeCompatTable}}

The **`activationStart`** read-only property represents the time between when a document starts prerendering and when it is activated.

## Value

> A number, representing the duration between document prerendering start and activation, in seconds.

Reviewer: Is it zero if the document is not prerendering? Does it only populate on activation?
```suggestion
A number, representing the duration between document prerendering start and activation, in seconds.
0 if the page has not prerendered or is still prerendering.
```

Author: Yup, I think you're right. The spec says "Every [Document](https://dom.spec.whatwg.org/#document) has an activation start time, which is initially a [DOMHighResTimeStamp](https://w3c.github.io/hr-time/#dom-domhighrestimestamp) with a time value of zero."

I've added this info.

## Examples

### Detecting prerendered pages

> When a prerendered document is activated, `activationStart` is set to a non-zero time. The following function can check for prerendering _and_ prerendered pages:

Reviewer: Text makes less sense here than in the prerendering page, where you have a bit more context.
```suggestion
When a prerendered document is activated, `activationStart` is set to a non-zero time. The following function can check whether a page is {{DOMxRef("Document.prerendering","prerendering")}} or has already prerendered.
```

Author: Looks good; change made. Thanks!

```js
function pagePrerendered() {
  return (
    document.prerendering ||
    self.performance?.getEntriesByType?.("navigation")[0]?.activationStart > 0
  );
}
```

### Measuring user-perceived performance milestones

> With prerendered pages, a page may have been created long before it was actually navigated to. When conducting performance measurements on prerendered pages, it is vital to use `activationStart` to avoid misleading measurements.

Reviewer: ```suggestion
With prerendered pages, a page may have been created long before it was actually navigated to. When conducting performance measurements on prerendered pages, it is vital to compare results to `activationStart` in order to avoid misleading measurements.
```

Author: Sounds great, thanks. I've pretty much used it as-is, but I've included a link to the Performance API landing page.

```js
// When the activation navigation started
let activationStart =
  performance.getEntriesByType("navigation")[0].activationStart;

> // When First Paint occurred

Reviewer: First pain a proper noun?
```suggestion
// Time to first paint
```

Author: Heh, updated. I copied this from an explainer somewhere ;-)

I also updated the first comment so it fits in with the suggested new style.

let firstPaint = performance.getEntriesByName("first-paint")[0].startTime;

> // When First Contentful Paint occurred

Reviewer: ```suggestion
// Time to first contentful paint
```

Author: Updated

---
title: "PerformanceNavigationTiming: activationStart property"
short-title: activationStart
slug: Web/API/PerformanceNavigationTiming/activationStart
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.PerformanceNavigationTiming.activationStart
---

{{APIRef("Performance API")}}{{SeeCompatTable}}

The **`activationStart`** read-only property represents the time between when a document starts prerendering and when it is activated.

## Value

> A {{domxref("DOMHighResTimeStamp")}} representing the duration between document prerendering start and activation in seconds.

Reviewer 2: ```suggestion
A {{domxref("DOMHighResTimeStamp")}} representing the duration between document prerendering start and activation in milliseconds.
```

Author: Updated

The value is `0` if the page has not prerendered or is still prerendering.

## Examples

### Detecting prerendered pages

> When a prerendered document is activated, `activationStart` is set to a non-zero time. The following function can check whether a page is {{DOMxRef("Document.prerendering","prerendering")}} or has already prerendered:

Author: Good call. I've updated it to "the current time'. I've also gotten rid of other similar instances of "non-zero time" in other documents.
